### PR TITLE
emit_x64{_vector}_floating_point: AVX{512} implementation of ZeroIfNaN

### PR DESCRIPTION
**Pull request by Wunkolo**
This PR primarily adds constants+functions for interfacing with the `vfixupimm*` instruction for a later pass of optimizing NaN-handling.
Currently it only finds usefulness optimizing `ZeroIfNaN` from two/three instructions into one `vfixupimm` instruction.

### CODE REVIEW

**Comment by Wunkolo** [comment]
Impulsively put `AVX2` when it's actually just the regular `AVX`. This PR also includes an `AVX` implementation for non-vector `ZeroIfNan`

File: src/dynarmic/backend/x64/emit_x64_floating_point.cpp [hunk 1]
```diff
@@ -116,9 +117,20 @@ void DenormalsAreZero(BlockOfCode& code, EmitContext& ctx, std::initializer_list
 
 template<size_t fsize>
 void ZeroIfNaN(BlockOfCode& code, Xbyak::Xmm xmm_value, Xbyak::Xmm xmm_scratch) {
-    code.xorps(xmm_scratch, xmm_scratch);
-    FCODE(cmpords)(xmm_scratch, xmm_value);  // true mask when ordered (i.e.: when not an NaN)
-    code.pand(xmm_value, xmm_scratch);
+    if (code.HasHostFeature(HostFeature::AVX512_OrthoFloat)) {
+        // Replace QNaN and SNaN with zero
+        constexpr u32 nan_to_zero = FixupLUT(
```
Comment (lioncash) [inline src/dynarmic/backend/x64/emit_x64_floating_point.cpp:122]:
```suggestion
        constexpr u32 nan_to_zero = FixupLUT(FpFixup::ZeroPos,
                                             FpFixup::ZeroPos);
```

Ditto for the occurrence below as well.

File: src/dynarmic/backend/x64/emit_x64_vector_floating_point.cpp [hunk 2]
```diff
@@ -203,7 +204,12 @@ void ForceToDefaultNaN(BlockOfCode& code, FP::FPCR fpcr, Xbyak::Xmm result) {
 template<size_t fsize>
 void ZeroIfNaN(BlockOfCode& code, Xbyak::Xmm result) {
     const Xbyak::Xmm nan_mask = xmm0;
-    if (code.HasHostFeature(HostFeature::AVX)) {
+    if (code.HasHostFeature(HostFeature::AVX512_OrthoFloat)) {
+        // Replace QNaN and SNaN with zero
```
Comment (MerryMage) [inline src/dynarmic/backend/x64/emit_x64_vector_floating_point.cpp:208]: ditto

File: src/dynarmic/backend/x64/emit_x64_floating_point.cpp [hunk 3]
```diff
@@ -116,9 +117,19 @@ void DenormalsAreZero(BlockOfCode& code, EmitContext& ctx, std::initializer_list
 
 template<size_t fsize>
 void ZeroIfNaN(BlockOfCode& code, Xbyak::Xmm xmm_value, Xbyak::Xmm xmm_scratch) {
-    code.xorps(xmm_scratch, xmm_scratch);
-    FCODE(cmpords)(xmm_scratch, xmm_value);  // true mask when ordered (i.e.: when not an NaN)
-    code.pand(xmm_value, xmm_scratch);
+    if (code.HasHostFeature(HostFeature::AVX512_OrthoFloat)) {
+        // Replace QNaN and SNaN with zero
```
Comment (MerryMage) [inline src/dynarmic/backend/x64/emit_x64_floating_point.cpp:121]: Unnecessary comment.

**Review by MerryMage — APPROVED** [review]

File: src/dynarmic/backend/x64/constants.h [hunk 4]
```diff
@@ -42,4 +43,46 @@ constexpr u8 b = 0b11001100;
...
+    OnePos = 0b1010,      // +1
+    Half = 0b1011,        // 0.5
+    Ninety = 0b1100,      // 90.0
+    PiHalf = 0b1101,      // PI /2
+    MaxPos = 0b1110,      // +{FLT_MAX,DBL_MAX}
+    MaxNeg = 0b1111,      // -{FLT_MAX,DBL_MAX}
+};
+
+// Generates 32-bit LUT for vfixupimm instruction
+constexpr u32 FixupLUT(
```
Comment (MerryMage) [inline src/dynarmic/backend/x64/constants.h:67]: The following is probably preferrable
```
constexpr u32 FixupLUT(FpFixup Src_QNaN = FpFixup::A,
                       FpFixup Src_SNaN = FpFixup::A,
                       FpFixup Src_Zero = FpFixup::A,
                       FpFixup Src_OnePos = FpFixup::A,
                       FpFixup Src_InfNeg = FpFixup::A,
                       FpFixup Src_InfPos = FpFixup::A,
                       FpFixup Src_Pos = FpFixup::A,
                       FpFixup Src_Neg = FpFixup::A) {
```

File: src/dynarmic/backend/x64/constants.h [hunk 5]
```diff
@@ -42,4 +43,46 @@ constexpr u8 b = 0b11001100;
 constexpr u8 c = 0b10101010;
 }  // namespace Tern
 
+// Opcodes for use with vfixupimm
+enum class FpFixup : u8 {
```
Comment (MerryMage) [inline src/dynarmic/backend/x64/constants.h:47]: Being picky, but would make more sense for signs to come before values: ```
enum class FpFixup : u8 {
    A = 0b0000,         // A
    B = 0b0001,         // B
    QNaN_B = 0b0010,    // QNaN with sign of B
    IndefNaN = 0b0011,  // Indefinite QNaN (Negative QNaN with no payload on x86)
    NegInf = 0b0100,    // -Infinity
    PosInf = 0b0101,    // +Infinity
    Inf_B = 0b0110,     // Infinity with sign of B
    NegZero = 0b0111,   // -0
    PosZero = 0b1000,   // +0
    NegOne = 0b1001,    // -1
    PosOne = 0b1010,    // +1
    Half = 0b1011,      // 0.5
    Ninety = 0b1100,    // 90.0
    HalfPi = 0b1101,    // PI/2
    PosMax = 0b1110,    // +{FLT_MAX,DBL_MAX}
    NegMax = 0b1111,    // -{FLT_MAX,DBL_MAX}
};
```

**Pull request merged**